root_orchestrator/cloud-scheduler/calculation.py

```diff
@@ -12,11 +12,21 @@ def calculate(job_id, job):
 
 
 def constraint_based_scheduling(job, constraints):
+    filtered_active_clusters = []
     for constraint in constraints:
         constraint_type = constraint.get("type")
         if constraint_type == "direct":
             return direct_service_mapping(job, constraint.get("cluster"))
-    return greedy_load_balanced_algorithm(job=job)
+        if constraint_type == "addons":
+            for cluster in cluster_operations.get_resources(active=True) or []:
+                if (
+                    cluster.get("supported_addons")
+                    and constraint.get("needs")
+                    and set(constraint.get("needs")).issubset(set(cluster.get("supported_addons")))
+                ):
+                    filtered_active_clusters.append(cluster)
+
+    return greedy_load_balanced_algorithm(job=job, active_clusters=filtered_active_clusters)
```

Comment:
Tested it in local cluster and an issue arised. 
If in SLA: `contraints` is not set, and therefore None, then this works.
If `contraints`: [] is an empty array, then `constraint_based_scheduling` is called and now calls `greedy_load_balanced_algorithm` with empty `filtered_active_clusters`. Can we quikly add the check that we call `constraint_based_scheduling` only when contraints is both != None and len(contraints)>0?

Reply:
Sure thing! Great catch. Will fix ASAP

Reply:
Is fixed